Simplified/NYPLBook.m

```diff
@@ -563,11 +563,11 @@ - (NYPLBookContentType)defaultBookContentType
     [finalTypes addObject:path.types.lastObject];
   }
 
+  NYPLBookContentType defaultType = NYPLBookContentTypeUnsupported;
   if (finalTypes.count == 1) {
-    return NYPLBookContentTypeFromMIMEType(finalTypes.firstObject);
+    defaultType = NYPLBookContentTypeFromMIMEType(finalTypes.firstObject);
```

Comment:
The logic for a single item in finalTypes seems the same as the logic for multiple items. Is it necessary to treat the one-item case differently?

Reply:
> It's valid OPDS for a book to have no acquisition paths -- the server need only provide a link. And sometimes we don't even provide a link. Right now, NYPL's production server knows not to tell the client about any PDFs, but this knowledge is hacked in and it does this by serving the OPDS entry for a book and omitting all of the links that would end up in a PDF.
> 
> If the book is only available as a PDF, this means the OPDS entry is served with no links at all. That's _not_ valid OPDS but the client should be prepared to handle it.

In this method, it's valid for there to be no _supported_ acquisition paths, because they got filtered out by `supportedAcquisitionPathsForAllowedTypes:...`. That's why I think it doesn't make sense to throw an exception (or even log anything), since this method was already sometimes returning `NYPLBookContentTypeUnsupported`. The only reason this didn't crash the app before is that we happened not to call `defaultBookContentType` if there weren't any valid acquisition paths.

Reply:
That makes sense.

Reply:
I think you're right about the single / multiple item cases. Thanks for pointing it out; it makes the code much simpler!

Reply:
The Android implementation also has separate but equivalent code for single-item lists, so it's possible that there used to be a difference that has since been removed.